File: frontend/types/evaluacionReportes.ts
// Tipos de los reportes de resultados de evaluaciones. Espejo de schemas/evaluacion_reportes.py.

export interface LoteEvaluacion {
  id: string
  empresa_id: string
  periodo: string
  importado_por: string | null
  created_at: string
}

export interface LotesResponse {
  items: LoteEvaluacion[]
  total: number
}

export interface ResumenCiclo {
  evaluados: number
  con_nota_final: number
  promedio: number | null
  nota_mas_alta: number | null
  nota_mas_baja: number | null
  evaluaciones: number
}

export interface BrechaItem {
  empleado_id: string | null
  apellido: string
  nombre: string
  auto: number | null
  terceros: number | null
  brecha: number | null
}

export interface SectorItem {
  sector: string
  evaluados: number
  promedio: number
  minima: number
  maxima: number
}

export interface CompetenciaItem {
  competencia: string
  promedio: number
  n: number
}

export interface CompetenciasReporte {
  lider: CompetenciaItem[]
  general: CompetenciaItem[]
  n_lider: number
  n_general: number
}

export interface MetricasResponse {
  resumen: ResumenCiclo
  brecha: BrechaItem[]
  sectores: SectorItem[]
  competencias: CompetenciasReporte
}

export interface EvaluadoListadoItem {
  id: string
  empleado_id: string | null
  apellido: string
  nombre: string
  sector: string | null
  superior: string | null
  tipos: string[]
  perfil: string
  nota_final: number | null
  asignado: boolean
}

export interface EvaluadoListadoResponse {
  items: EvaluadoListadoItem[]
  total: number
}

export interface FichaResponse {
  apellido: string
  nombre: string
  sector: string | null
  perfil: string
  nota_final: number | null
  competencias: string[]
  tipos: string[]
  celdas: Record<string, Record<string, number>>
  promedio_terceros: Record<string, number>
}

File: backend/schemas/evaluacion_reportes.py
"""
Schemas de los reportes de un lote de evaluaciones importado (fase 5.2): métricas del ciclo,
listado filtrable de evaluados y ficha individual. Todo se computa en Python (ver
_evaluacion_metricas); acá solo la forma de salida.
"""
from typing import Dict, List, Optional

from pydantic import BaseModel


# ── Métricas del ciclo ────────────────────────────────────────────────────────

class ResumenCiclo(BaseModel):
    evaluados: int
    con_nota_final: int
    promedio: Optional[float] = None
    nota_mas_alta: Optional[float] = None
    nota_mas_baja: Optional[float] = None
    evaluaciones: int                      # pares (evaluado × tipo de evaluador)


class BrechaItem(BaseModel):
    empleado_id: Optional[str] = None
    apellido: str
    nombre: str
    auto: Optional[float] = None           # promedio de sus autoevaluaciones
    terceros: Optional[float] = None       # promedio de los demás tipos
    brecha: Optional[float] = None         # auto − terceros (None si falta un lado)


class SectorItem(BaseModel):
    sector: str
    evaluados: int
    promedio: float
    minima: float
    maxima: float


class CompetenciaItem(BaseModel):
    competencia: str
    promedio: float
    n: int                                 # cantidad de notas promediadas


class CompetenciasReporte(BaseModel):
    # Dos rankings SEPARADOS: líder y general nunca se mezclan (sets de competencias distintos).
    lider: List[CompetenciaItem]
    general: List[CompetenciaItem]
    n_lider: int                           # evaluados de perfil líder
    n_general: int


class MetricasResponse(BaseModel):
    resumen: ResumenCiclo
    brecha: List[BrechaItem]
    sectores: List[SectorItem]
    competencias: CompetenciasReporte


# ── Listado + ficha ───────────────────────────────────────────────────────────

class EvaluadoListadoItem(BaseModel):
    id: str                                # id de la fila (evaluacion_evaluados) — abre la ficha
    empleado_id: Optional[str] = None
    apellido: str
    nombre: str
    sector: Optional[str] = None
    superior: Optional[str] = None
    tipos: List[str]
    perfil: str
    nota_final: Optional[float] = None
    asignado: bool                         # False = no se matcheó a un empleado


class EvaluadoListadoResponse(BaseModel):
    items: List[EvaluadoListadoItem]
    total: int


class FichaResponse(BaseModel):
    apellido: str
    nombre: str
    sector: Optional[str] = None
    perfil: str
    nota_final: Optional[float] = None
    competencias: List[str]                          # en el orden del archivo
    tipos: List[str]                                 # solo los presentes, en orden canónico
    celdas: Dict[str, Dict[str, float]]              # competencia → {tipo: nota}; ausente = no aplica
    promedio_terceros: Dict[str, float]              # competencia → promedio de terceros
